### ISSUE TITLE
Pylint false positive when unpacking sys.argv

### ISSUE THREAD

**Issue by rvagner42** [PyCQA/pylint#3923]
### Steps to reproduce
Call pylint on the following code:

```python
import sys


def do_something(a, b):
    """
    It doesn't matter what this function does, nor how many parameters it takes.
    """
    print(a + b)


if __name__ == "__main__":
    do_something(*sys.argv)
```

### Current behavior
```
script.py:12:4: E1120: No value for argument 'a' in function call (no-value-for-parameter)
script.py:12:4: E1120: No value for argument 'b' in function call (no-value-for-parameter)
```

### Expected behavior
No error as I make sure in my real script that `sys.argv` has enough values. 

### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.8.1 (tags/v3.8.1:1b293b6, Dec 18 2019, 23:11:46) [MSC v.1916 64 bit (AMD64)]
```
---
As a side note, this works properly when using python 2.7.18 and pylint 1.9.5

**Comment by hippo91** [comment]
@rvagner42 thanks for this report.
`sys.argv` is dynamically populated and `pylint` is just a static analyzer, that's why it doesn't know how many arguments are going to be passed into the function call. That's why it emits such messages.
